typedoc.json

```diff
@@ -0,0 +1,33 @@
+{
+	"$schema": "https://typedoc.org/schema.json",
+	"categoryOrder": [
+		"Node Types",
+		"Nodes - Type Guards",
+		"Nodes - Flag Utilities",
+		"Nodes - Other Utilities",
+		"Modifier Utilities",
+		"Scope Utilities",
+		"Syntax Utilities",
+		"Compiler Options",
+		"*",
+		"Other"
+	],
+	"cleanOutputDir": true,
+	"defaultCategory": "Other",
+	"entryPointStrategy": "resolve",
+	"entryPoints": ["./src/index.ts"],
+	"exclude": ["./**/test/", "./**/*.test.*"],
+	"excludeExternals": true,
+	"excludeInternal": true,
+	"excludeNotDocumented": false,
+	"excludePrivate": true,
+	"hideGenerator": true,
+	"includeVersion": true,
+	"out": "./docs",
+	"plugin": [
+		"typedoc-plugin-coverage",
+		"typedoc-plugin-mdn-links",
+		"typedoc-plugin-versions"
+	],
+	"readme": "none"
+}
```

Comment:
[Comment] I have a few miscellaneous feature requests that are definitely not blocking for this PR and probably shouldn't go in it even if they're immediately actionable:

* Integrating [Konamimojisplosion](https://github.com/JoshuaKGoldberg/Konamimojisplosion) 😁 
* Adding a <kbd>Ctrl</kbd>/<kbd>Cmd</kbd>+<kbd>K</kbd> shortcut for search the way Docusaurus and a lot of other sites / site generators have
* Adding lint rules around tags: avoiding blank/redundant ones and including useful on exported things

...just posting here so I don't forget them.